File: gpfs-app/src/main/resources/static/js/gpfs/gpfs/gpfs.service.js
angular.module('AdminUI')
    .service('GpfsService', GpfsService);

function GpfsService($resource) {
    return $resource('/gpfs', null, {
    	findByCompanyAndYear: {
    		url: '/gpfs/:companyId/:year',
    		method: 'GET',
    		isArray: false
    	}
    });
}

File: gpfs-gpfs/src/main/java/com/gpfs/gpfs/service/GpfsService.java
package com.gpfs.gpfs.service;

import java.util.Optional;

import com.gpfs.core.service.GpfsJpaService;
import com.gpfs.gpfs.Gpfs;
import com.gpfs.gpfs.service.custom.GpfsServiceCustom;

/**
 *
 * @author Mark Martinez on Aug 5, 2017
 *
 */
public interface GpfsService extends GpfsServiceCustom, GpfsJpaService<Gpfs> {

	Optional<Gpfs> findByCompanyIdAndYear(Long companyId, int year);

}
